pkg/kubelet/config/config.go

```diff
@@ -413,10 +425,19 @@ func podsDifferSemantically(existing, ref *api.Pod) bool {
 
 // checkAndUpdatePod updates existing, and:
 //   * if ref makes a meaningful change, returns needUpdate=true
+//   * if ref makes a meaningful change, and this change is graceful deletion, returns needGracefulDelete=true
 //   * if ref makes no meaningful change, but changes the pod status, returns needReconcile=true
-//   * else return both false
-//   Now, needUpdate and needReconcile should never be both true
-func checkAndUpdatePod(existing, ref *api.Pod) (needUpdate, needReconcile bool) {
+//   * else return all false
+//   Now, needUpdate, needGracefulDelete and needReconcile should never be both true
+func checkAndUpdatePod(existing, ref *api.Pod) (needUpdate, needReconcile, needGracefulDelete bool) {
+	// 1. this is an graceful delete
+	if ref.DeletionTimestamp != nil && !reflect.DeepEqual(existing.DeletionTimestamp, ref.DeletionTimestamp) {
```

Comment:
Although pod won't be updated after `Deletiontimestamp` is set, but I still think it would be safer to do less change, such as:
```
if deletiontimestamp != nil {
  needGracefulDelete = true
} else {
  needUpdate = true
}
```